client/components/Item/RelatedItems.vue

```diff
@@ -89,39 +89,54 @@ export default Vue.extend({
       loading: true
     };
   },
-  watch: {
-    item(): void {
-      this.refreshItems();
-    }
+  computed: {
+    ...mapGetters('items', ['getItems', 'getMissingIds'])
   },
-  beforeMount() {
-    try {
-      this.refreshItems();
-    } catch (error) {
-      this.pushSnackbarMessage({
-        message: this.$t('unableGetRelated'),
-        color: 'error'
-      });
+  watch: {
+    'item.Id': {
+      immediate: true,
+      async handler(): Promise<void> {
+        await this.refreshItems();
+      }
     }
   },
   methods: {
     ...mapActions('snackbar', ['pushSnackbarMessage']),
     async refreshItems(): Promise<void> {
-      this.loading = true;
-
       if (this.item.Id) {
-        const response = await this.$api.library.getSimilarItems({
-          itemId: this.item.Id,
-          userId: this.$auth.user?.Id,
-          limit: this.vertical ? 5 : 12
-        });
+        try {
+          this.loading = true;
 
-        if (response.data.Items) {
-          this.relatedItems = response.data.Items;
+          const response = await this.$api.library.getSimilarItems({
+            itemId: this.item.Id,
+            userId: this.$auth.user?.Id,
+            limit: this.vertical ? 5 : 12
+          });
+
+          if (response.data.Items) {
+            const itemIds: string[] = response.data.Items.map(
+              (item: BaseItemDto) => {
+                return item.Id as string;
+              }
+            );
+
+            const missingIds = this.getMissingIds(itemIds);
+
+            if (missingIds.length > 0) {
+              await this.$items.getItems({ ids: missingIds });
+            }
+
+            this.relatedItems = this.getItems(itemIds);
```

Comment:
This entire section seems a bit convoluted (Mainly because of the API, but still).

We're essentially fetching the items twice here: once via `getSimilarItems` and then re-fetching the ones not in the store via `getItems`.

So we fetch a bunch of items... just to discard them completely. At that point, we may as well add them all to the store without any filtering.

Reply:
Aren't the items from the first query incomplete?

Reply:
You can pretty much always specify which fields you want.

Reply:
In that case, can't we in the first step only ask for IDs and then do the second step as here?

Reply:
No, because you always get some info on the item. Plus the server will still have to run the queries to get what you want.

The issue here is doing 2 API calls, with all that implies: two handshakes, two requests, two DB queries, etc.